app/controllers/items_controller.rb

```diff
@@ -1,7 +1,27 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
+  before_action :set_wishlist, only: %i(new create)
   before_action :set_wishlist_and_item, only: %i(edit update)
```

Comment:
If you're extracting `set_wishlist` as another method, then change `set_wishlist_and_item` to just `set_item` and run both `set_wishlist` and `set_item` for the methods that need them both.